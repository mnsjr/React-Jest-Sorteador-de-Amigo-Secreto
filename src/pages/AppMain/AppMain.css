.container-app-main {
    width: 100%;
}