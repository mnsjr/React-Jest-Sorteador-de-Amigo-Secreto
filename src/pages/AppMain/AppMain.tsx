// CSS
import './AppMain.css';

// COMPONENTS
import Formulario from '../../componentes/Formulario/Formulario';
import Cabecalho from '../../componentes/Cabecalho/Cabecalho';

// REACT
import React from 'react'

const AppMain = () => {
  return (
    <div className="App">
        
        <header className="App-header">
            <Cabecalho/>
            <Formulario />
        </header>
    </div> 
  )
}

export default AppMain