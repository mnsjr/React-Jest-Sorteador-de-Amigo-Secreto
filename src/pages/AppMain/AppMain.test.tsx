import { render, screen } from "@testing-library/react";
import React from "react";
import AppMain from "./AppMain";
import { RecoilRoot } from "recoil";

const mockNavegacao = jest.fn()

jest.mock('react-router-dom', () => {
    return {
        useNavigate: () => mockNavegacao
    }
})

describe('a pagina de config', () => {
    test('renders learn react link', () => {
        render(
            <RecoilRoot>
            <AppMain />
            </RecoilRoot>); 
        // THIS TEST IS LOOKING FOR SOME TEXT IN APP COMPONENT, THE SAME CAN DO WITH ANY COMPONENT AND TEXT
        const linkElement = screen.getByText(/Vamos começar!/i);
        expect(linkElement).toBeInTheDocument();
    });

    test('deve ser renderizada corretamente', () => {
        const { container } = render(<RecoilRoot>
            <AppMain />
        </RecoilRoot>)

        // ESTA É UMA PÁGINA MERAMENTE VISUAL, 
        // ENTÃO IREMOS FAZER UM TESTE DE SNAPSHOT, 
        // COMPARANDO A IMAGEM DA TELA PARA GARANTIR QUE ELA SEMPRE SERÁ RENDERIZADA DA MESMA FORMA

        // TO UPDATE SNAPSHOT TEST:
        // Press u to update failing snapshots.
        expect(container). toMatchSnapshot()
    })
})