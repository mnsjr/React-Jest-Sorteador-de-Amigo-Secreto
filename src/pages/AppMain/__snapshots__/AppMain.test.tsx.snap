// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de config deve ser renderizada corretamente 1`] = `
<div>
  <section
    class="card"
  >
    <div
      class="page-form"
    >
      <div
        class="form-title"
      >
        <div
          class="form-title-dice"
        >
          <img
            alt="Imagem decorativa na figura de um dado."
            src="dice.svg"
          />
          <img
            alt="Imagem decorativa na figura de um dado."
            src="dice2.svg"
          />
          <img
            alt="Imagem decorativa na figura de um dado."
            src="dice.svg"
          />
        </div>
        <h1>
          SORTEADOR DE AMIGO SECRETO
        </h1>
      </div>
      <div
        class="page-form-img"
      >
        <img
          alt="Imagem decorativa, mulher jovem, parda, cabelo encaracolado estilo black power, vestindo uma blusa laranja com um tablet amarelo nas mãos."
          src="participante.png"
        />
      </div>
    </div>
    <section
      class="container-app-main"
    >
      <h2>
        Vamos começar!
      </h2>
      <form>
        <div
          class="grupo-input-btn"
        >
          <div
            class="grupo-input-btn-1"
          >
            <input
              placeholder="Insert players names"
              type="text"
              value=""
            />
            <button
              disabled=""
            >
              Adicionar
            </button>
          </div>
          <div
            class="grupo-input-btn-2"
          />
        </div>
      </form>
      <ul
        class="lista-participantes-list"
      />
      <footer
        class="rodape-configuracoes"
      >
        <button
          class="botao"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Imagem ilustrativa."
          src="sacolas.png"
        />
      </footer>
    </section>
  </section>
</div>
`;
